src/Data/ByteString/Base64.hs

```diff
@@ -107,8 +110,8 @@ decodeBase64 bs@(PS _ _ !l)
 -- >>> decodebase64Lenient "U3V="
 -- "Su"
 --
-decodeBase64Lenient :: ByteString -> ByteString
-decodeBase64Lenient = decodeBase64Lenient_ decodeB64Table
+decodeBase64Lenient :: Base64 'StdUnpadded ByteString -> ByteString
+decodeBase64Lenient = decodeBase64Lenient_ decodeB64Table . extractBase64
```

Comment:
Here is where it starts getting wonky - this type is now incorrect, since `*Lenient` also works for padded strings. The type would ideally look more like 

```haskell
decodeBase64Lenient 
  :: Base64 ('StdUnpadded \/ 'StdPadded) ByteString
  -> ByteString
```

where `\/` is a union of the two types. If we can do that, then it actually is a really strong specification that I like alot. However, it also begs the question of whether or not this should just stay `ByteString -> ByteString`, since you aren't necessarily giving it RFC-compliant strings. So perhaps it would look more like 

```haskell
decodeBase64Lenient 
  :: Base64 ('StdUnpadded \/ 'StdPadded \/ 'NonStandard) ByteString
  -> ByteString
```

And we should add a `NonStandard` case to `Alphabet` to cover wonky encodings that people can just assert.

Reply:
Resolved in https://github.com/emilypi/base64/commit/e2bdbe14244b7542a4d68885ce41ead426dccf54